lib/maths/CTimeSeriesChangeDetector.cc

```diff
@@ -297,8 +298,8 @@ void CUnivariateTimeSeriesChangeDetector::addSamples(const TTimeDoublePr1Vec& sa
         core_t::TTime time{samples[i].first};
         weight = weights[i];
         maths_t::setWinsorisationWeight(winsorisation::MINIMUM_WEIGHT, weight);
-        m_TrendModel->addPoint(
-            time, CBasicStatistics::mean(m_TrendModel->value(time, 0.0)), weight);
+        m_TrendModel->addPoint(time, CBasicStatistics::mean(m_TrendModel->value(time, 0.0)),
+                               onModelChange, weight);
```

Comment:
I don't think we need this here. We don't add seasonal components to the decomposition while we're testing for a change. If this was an optional argument, as suggested above, this code could just stay as it is.